### PR TITLE
UCT/CUDA: use only driver API for memory detection

### PR DESCRIPTION
**Pull request by bureddy**
- removed fallback check with runtime API, which is causing overhead in multiple context scenario.
      This seems not necessary anymore with latest drivers. And also memory detection with UCTs will be replaced with UCP level ptr caching.
   -  common code moved to new base file

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/6473/ for details (Mellanox internal link).

**Comment by shamisp** [comment]
if this is driver version depended, can we add a check for driver version ? If it is not up-to-date we should issue warning or abort ?

File: src/uct/cuda/base/cuda_md.c [hunk 1]
```diff
@@ -0,0 +1,27 @@
+/**
+ * Copyright (C) Mellanox Technologies Ltd. 2018.  ALL RIGHTS RESERVED.
+ * See file LICENSE for terms.
+ */
+
+#include <uct/cuda/base/cuda_md.h>
+
+int uct_is_cuda_mem_type_owned(uct_md_h md, void *addr, size_t length)
```
Comment (yosefe) [inline src/uct/cuda/base/cuda_md.c]: uct_cuda_is_mem_type_owned

File: src/uct/cuda/base/cuda_md.c [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    CUresult cu_err;
+
+    if (addr == NULL) {
+        return 0;
+    }
+
+    cu_err = cuPointerGetAttribute(&memory_type,
+                                   CU_POINTER_ATTRIBUTE_MEMORY_TYPE,
+                                   (CUdeviceptr)addr);
+    if (cu_err == CUDA_SUCCESS) {
```
Comment (yosefe) [inline src/uct/cuda/base/cuda_md.c]: return  (cu_err == CUDA_SUCCESS) && (memory_type == CU_MEMORYTYPE_DEVICE);

File: src/uct/cuda/base/cuda_md.h [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+ * See file LICENSE for terms.
+ */
+
+#ifndef UCT_CUDA_MD_H
+#define UCT_CUDA_MD_H
+
+#include <uct/base/uct_md.h>
+
+#include <cuda_runtime.h>
+#include <cuda.h>
```
Comment (yosefe) [inline src/uct/cuda/base/cuda_md.h]: do we really need to include cuda headers here?

**Comment by bureddy** [comment]
@shamisp No, changes are not driverversion dependent.

**Comment by bureddy** [comment]
@yosefe fixed comments

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/6483/ for details (Mellanox internal link).

**Pull request merged**